android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BeaconConfigFragment.java

```diff
@@ -327,21 +335,28 @@ private void saveEditCardUrlToBeacon() {
     if (!URLUtil.isNetworkUrl(url)) {
       url = "http://" + url;
     }
+    // Create the callback object to set the url
+    UrlShortener.ModifiedUrlCallback urlSetter = new UrlShortener.ModifiedUrlCallback() {
+      public void onNewUrl(String newUrl){
+        // Write the url to the device
+        try {
+          // Note: setting txPower here only really updates txPower for v1 beacons
+          ConfigUriBeacon configUriBeacon = new ConfigUriBeacon.Builder()
+              .txPowerLevel(TX_POWER_DEFAULT)
+              .uriString(newUrl)
+              .build();
+          mUriBeaconConfig.writeUriBeacon(configUriBeacon);
+        } catch (URISyntaxException e) {
+          e.printStackTrace();
+          Toast.makeText(getActivity(), R.string.config_url_error, Toast.LENGTH_SHORT).show();
```

Comment:
Does anything bad can happen if the user closes the view / activity before the callback is called?